### PR TITLE
docs: add namespace guide

### PR DESCRIPTION
**Pull request by sixhobbits**
**What this PR does / why we need it**:

- adds a guide on how to use namespaces

**Which issue(s) this PR fixes**:

- n/a

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/guides/namespaces.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Environments and Namespaces
+
+Every Garden project has one or more environments that are defined in the project level Garden configuration. Teams often define environments such as `dev`, `ci`, and `prod`. 
+
+Each environment can be broken down into several "namespaces", and each Garden run operates in a specific namespace. (This is not to be confused with a Kubernetes Namespace resource, although you will often use the same name for your Garden Namespace and your Kubernetes Namespace.)
+
+To specify which Garden Namespace to use, you can use either of the following:
```
Comment (eysi09) [inline docs/guides/namespaces.md:7]: All references to Garden namespaces should be lower case. So change "Namespace" to "namespace" here and below. 

However, a Kubernetes "Namespace" resource is capitalised :)

File: docs/guides/namespaces.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  - name: preview
+    defaultNamespace: ${var.prev-env-name}
+    variables:
+      hostname: ${var.prev-env-name}.preview.<my-company>.com
+  - name: dev
+    defaultNamespace: ${var.dev-env-name}
+    variables:
+      hostname: ${var.dev-env-name}.dev.<my-company>.com
+  - name: prod
+		defaultNamespace: my-project
```
Comment (eysi09) [inline docs/guides/namespaces.md:61]: These white spaces look odd.

File: docs/guides/namespaces.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  - name: kubernetes
+    namespace: ${environment.namespace} # <--- Ensure the K8s namespace matches the Garden namespace
+    defaultHostname: ${var.hostname}
+    # ...
+```
+
+This allows each developer to get a unique namespace and a unique hostname for each service. Some further notes:
+
+* The `dev-env-name` namespace will be something like `my-project-janedoe` so each developer has a unique namespace per project.
+* This namespace is set at a provider level, so Garden will always deploy the project to that namespace when that developer runs `garden deploy`.
```
Comment (eysi09) [inline docs/guides/namespaces.md:75]: I think we can skip this point.

File: docs/guides/namespaces.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# Environments and Namespaces
+
+Every Garden project has one or more environments that are defined in the project level Garden configuration. Teams often define environments such as `dev`, `ci`, and `prod`. 
+
+Each environment can be broken down into several "namespaces", and each Garden run operates in a specific namespace. (This is not to be confused with a Kubernetes Namespace resource, although you will often use the same name for your Garden Namespace and your Kubernetes Namespace.)
+
+To specify which Garden Namespace to use, you can use either of the following:
+
+- Set a specific Namespace using the CLI with the `--env` flag and prepending the Namespace to the environment name using the following format `--env <namespace>.<environment>`
+- Specify the default Namespace in your Garden configuration file, using the `defaultNamespace` field under the `environments` specification.
```
Comment (eysi09) [inline docs/guides/namespaces.md:10]: Can you add a link to the `defaultNamespace`, like so:

using the \[\`defaultNamespace\`\](https://docs.garden.io/reference/project-config#environments-.defaultnamespace) field under the...

File: docs/guides/namespaces.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    namespace: ${environment.namespace} # <--- Ensure the K8s namespace matches the Garden namespace
+    defaultHostname: ${var.hostname}
+    # ...
+```
+
+This allows each developer to get a unique namespace and a unique hostname for each service. Some further notes:
+
+* The `dev-env-name` namespace will be something like `my-project-janedoe` so each developer has a unique namespace per project.
+* This namespace is set at a provider level, so Garden will always deploy the project to that namespace when that developer runs `garden deploy`.
+* The `base-hostname` variable gives a unique hostname per project and per developer too. For example, if Jane Doe is working on a project called "Phoenix Project" at a company called Acme, her hostname would be `phoenix-project-janedoe.acme.com`
```
Comment (eysi09) [inline docs/guides/namespaces.md:76]: Change to:

The `hostname` variable can be re-used in the module configuration. When using the `container` module type, you can e.g. set `hostname: my-service.${var.hostname}` under the [`services.ingress`](https://docs.garden.io/reference/module-types/container#services-.ingresses) field. A similar approach can be used for other module types.

File: docs/guides/namespaces.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+* This can also be used in the container module, to add another level of subdomains for individual services. So if the "Phoenix Project" includes a Redis database, the hostname could be `redis.phoenix-project.janedoe.acme.com`. This also works for Kubernetes modules.
```
Comment (eysi09) [inline docs/guides/namespaces.md:77]: Skip this.

**Review by eysi09 — CHANGES_REQUESTED** [review]
Great work @sixhobbits! Just had a few suggestions but overall looks good.

**Comment by sixhobbits** [comment]
@eysi09 - changes are done

**Review by eysi09 — APPROVED** [review]
Great work @sixhobbits! 🚀

**Pull request merged**